libcst/codemod/commands/convert_precent_format_to_fstring.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# pyre-strict
+import re
+from typing import Callable, cast
+
+import libcst as cst
+import libcst.matchers as m
+from libcst.codemod import VisitorBasedCodemodCommand
+
+
+USE_FSTRING_SIMPLE_EXPRESSION_MAX_LENGTH = 30
+
+
+def _match_simple_string(node: cst.CSTNode) -> bool:
+    if isinstance(node, cst.SimpleString) and not node.prefix.lower().startswith("b"):
+        # SimpleString can be a bytes and fstring don't support bytes
+        return re.fullmatch("[^%]*(%s[^%]*)+", node.raw_value) is not None
+    return False
+
+
+def _gen_match_simple_expression(module: cst.Module) -> Callable[[cst.CSTNode], bool]:
+    def _match_simple_expression(node: cst.CSTNode) -> bool:
+        # either each element in Tuple is simple expression or the entire expression is simple.
+        if (
+            isinstance(node, cst.Tuple)
+            and all(
+                len(module.code_for_node(elm.value))
+                < USE_FSTRING_SIMPLE_EXPRESSION_MAX_LENGTH
+                for elm in node.elements
+            )
+        ) or len(module.code_for_node(node)) < USE_FSTRING_SIMPLE_EXPRESSION_MAX_LENGTH:
+            return True
+        return False
+
+    return _match_simple_expression
+
+
+class EscapeStringQuote(cst.CSTTransformer):
+    def __init__(self, quote: str) -> None:
+        self.quote = quote
+        super().__init__()
+
+    def leave_SimpleString(
+        self, original_node: cst.SimpleString, updated_node: cst.SimpleString
+    ) -> cst.SimpleString:
+        # prefix, quote, innards = _string_prefix_and_quotes(original_node.value)
+        if self.quote == original_node.quote:
+            for quo in ["'", '"', "'''", '"""']:
+                if quo != original_node.quote and quo not in original_node.raw_value:
```

Comment:
Should we also check the literal value of the string before and after the change, just for safety?

Reply:
Checked and raise exception when the literal value isn't the same.